treeherder/model/derived/jobs.py

```diff
@@ -1383,6 +1383,11 @@ def _load_ref_and_job_data_structs(
                 'option_collection_hash': option_collection_hash
             })
 
+        if created is True:
+            # A new ReferenceDataSignature has been added
+            for item in ExclusionProfile.objects.all():
+                item.update_flat_exclusions()
```

Comment:
I think it would be better to override the ReferenceDataSignatures class's save method, as that will guarantee that this gets run when an objects gets saved. You can see lots of existing examples of this in `treeherder/model/models.py`, just search for "save".

Other than that, this *looks* right to me, though I'm not 100% confident in the logic of this code (I've only touched it a few times) -- a unit test would be helpful, I'm also going to try to get @camd to look at this to make sure you're on the right track.